controller.go

```diff
@@ -268,7 +268,7 @@ func (c *controller) SetClusterProvider(provider cluster.Provider) {
 	}
 	c.Unlock()
 
-	if provider == nil || sameProvider {
+	if provider == nil || (sameProvider && c.getAgent() != nil) {
```

Comment:
if I understood well, basically when the node leave the cluster never reset the cluster provider. If so, I will find much better to just reset the cluster provider to nil when the node leave the cluster. 
That way sameProvider will be false and the init will continue. What do you think? is it my understanding correct?

Reply:
That might work as well. I'll test it out and see!

Reply:
Thanks!

Reply:
Not sure if you review moby or not but I've made a PR over there that fixes it in a different way. 
https://github.com/moby/moby/pull/38626